types/react-cache/index.d.ts

```diff
@@ -0,0 +1,18 @@
+// Type definitions for react-cache 2.0
+// Project: https://github.com/facebook/react/tree/master/packages/react-cache
+// Definitions by: Spencer Miskoviak <https://github.com/skovy>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export interface Thenable<Value> {
+  then(resolve: (value: Value) => any, reject: (reason: any) => any): any;
+}
+
+export interface Resource<Input, Value> {
+  read(key: Input): Value;
+  preload(key: Input): void;
+}
+
+export function unstable_createResource<Input, Value>(
+  fetch: (input: Input) => Thenable<Value>,
+  maybeHashInput?: (input: Input) => string | number
```

Comment:
Interesting that in the Flow types they have this as a generic input, but don't actually use its genericness.

In TS, that being generic would have the effect of inferring literal types for the return, but that doesn't seem productive for a hash function either.